modules/polyfills/src/fetch-node/read-file.node.js

```diff
@@ -30,9 +30,15 @@ export async function readFile(url, options = {}) {
 
   if (isRequestURL(url)) {
     return new Promise((resolve, reject) => {
-      options = {...new URL(url), ...options};
+      const _url = new URL(url);
+      const _options = {
```

Comment:
My main concern is that we are removing the ability for the user to pass through additional options to `request`. Even more importantly, this way the user cannot override the `GET` method. 

Can we make something more flexible using `Object.assign`?

```
  const requestOptions = Object.assign({},
     new URL(url),
     // or
     { 
        hostname: _url.hostname,
        path: _url.pathname,
        method: 'GET',
      }, 
       options,
       options.fetch, 
      {
        headers: options.headers || {} 
      };
```